geoips/interfaces/base.py

```diff
@@ -319,6 +323,84 @@ def __new__(cls):
 
         return super(BaseInterface, cls).__new__(cls)
 
+    @abc.abstractmethod
+    def get_plugin(self, name, rebuild_registries=rbr):
+        """Abstract function for retrieving a plugin under a certain interface.
+
+        Parameters
+        ----------
+        name: str or tuple(str)
+            - The name of the yaml-based plugin. Either a single string or a tuple of
+              strings for product plugins.
+        rebuild_registries: boolean (default=False)
+            - Whether or not to rebuild the registries if get_plugin fails. If set to
+              true and get_plugin fails, rebuild the plugin registry, call then call
+              get_plugin once more with rebuild_registries toggled off, so it only gets
+              rebuilt once.
+        """
+        pass
+
+    def call_create_plugin_registries(self, yaml=False):
+        """Run command 'create_plugin_registries' as a subprocess.
+
+        This should be done whenever a plugin is attempted to be loaded but fails
+        because it isn't an entry in the plugin registry. If the plugin still cannot be
+        found after this function is ran, then report the issue to the user.
+
+        Parameters
+        ----------
+        yaml: boolean (default=False)
+            - Truth value as to whether or not we want to create a .yaml registry. If
+              false, then create the default .json registry instead.
+        """
+        args = ["create_plugin_registries"]
+        if yaml:
+            args += ["-s", "yaml"]
+        call(args, shell=False)
+        # Reload the interface's plugin_registry_module so that the plugin registry is
+        # in the most recent state.
+        reload(self.plugin_registry_module)
+        self.plugin_registry = self.plugin_registry_module.plugin_registry
+
+    def retry_get_plugin(self, name, rebuild_registries, err_str, err_type=PluginError):
+        """Re-run self.get_plugin, but call 'create_plugin_registries' beforehand.
+
+        By running 'create_plugin_registries', we automate the registration of plugins
+        in GeoIPS. If the plugin persists not to be found, then we'll raise an
+        appropriate PluginError as denoted by 'err_str'.
+
+        Parameters
+        ----------
+        name: str or tuple(str)
+            - The name of the yaml plugin. Either a single string or a tuple of strings
+              for product plugins.
+        rebuild_registries: boolean (default=False)
```

Comment:
This does not actually have a default at this point.

Reply:
I like having the methods themselves default to False, and the overall environment variable default to True, FWIW.